public/js/selfoss-db.js

```diff
@@ -1,6 +1,154 @@
 /**
  * db functions: client data repository
  */
+
+
+selfoss.dbOnline = {
```

Comment:
As I see it, these three objects form the model layer. Optimally, the `dbOnline` object should only take care of fetching/uploading data from the server. It could then be called `synchronizer`. The `dbOffline` should only be a nice wrapper above `dexie`, I would call it `storage`. Last, the `db` object could be an envelope above `synchronizer` and `storage`, providing an opaque model for the rest of the app. What I find the most important, none of the object should have any access to DOM.